From fe99b8ec45c64ba4deacd05ca7ca0e53a90f3ffc Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Thu, 4 Aug 2016 09:50:54 -0400
Subject: [PATCH 0727/1722] drm/amd/dal: Fallback to 8-bit linebuffer if not
 enough taps

If we can't do at least 2 taps try set linebuffer to 24bpp and
try taps validation again.

Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_resource.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c b/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c
index 98a2ecd..67d4b6b 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_resource.c
@@ -474,6 +474,14 @@ bool resource_build_scaling_params(
 	res = pipe_ctx->xfm->funcs->transform_get_optimal_number_of_taps(
 		pipe_ctx->xfm, &pipe_ctx->scl_data, &surface->scaling_quality);
 
+	if (!res) {
+		/* Try 24 bpp linebuffer */
+		pipe_ctx->scl_data.lb_bpp = LB_PIXEL_DEPTH_24BPP;
+
+		res = pipe_ctx->xfm->funcs->transform_get_optimal_number_of_taps(
+			pipe_ctx->xfm, &pipe_ctx->scl_data, &surface->scaling_quality);
+	}
+
 	dal_logger_write(pipe_ctx->stream->ctx->logger,
 				LOG_MAJOR_DCP,
 				LOG_MINOR_DCP_SCALER,
-- 
2.7.4

